src/osemgrep/cli_ci/Ci_subcommand.ml

```diff
@@ -144,42 +145,70 @@ let run (conf : Ci_CLI.conf) : Exit_code.t =
             (* TODO: metadata_dict["is_sca_scan"] = supply_chain *)
             (* TODO: proj_config = ProjectConfig.load_all()
                metadata_dict = {**metadata_dict, **proj_config.to_dict()} *)
-            let scan_id =
+            match
               Scan_helper.start_scan ~dry_run:conf.dryrun ~token
                 Semgrep_envvars.v.semgrep_url metadata_dict
-            in
-            let at_url_maybe ppf () =
-              if
-                Uri.equal Semgrep_envvars.v.semgrep_url
-                  (Uri.of_string "https://semgrep.dev")
-              then Fmt.string ppf ""
-              else
-                Fmt.pf ppf " at %a"
-                  Fmt.(styled `Bold string)
-                  (Uri.to_string Semgrep_envvars.v.semgrep_url)
-            in
-            Logs.app (fun m ->
-                m "  Fetching configuration from Semgrep Cloud Platform%a"
-                  at_url_maybe ());
-            let rules =
-              (* TODO: set sca to metadata.is_sca_scan / supply_chain *)
-              Scan_helper.fetch_scan_config ~token ~sca:false
-                ~dry_run:conf.dryrun ~full_scan:metadata.is_full_scan
-                metadata.repository
-            in
-            ( Some scan_id,
-              [
-                Common2.with_tmp_file ~str:rules ~ext:"json" (fun file ->
-                    let file = Fpath.v file in
-                    let res =
-                      Rule_fetching.load_rules_from_file ~registry_caching:false
-                        file
+            with
+            | Error msg ->
+                Logs.err (fun m -> m "%s" msg);
+                Error Exit_code.fatal
+            | Ok scan_id -> (
+                let at_url_maybe ppf () =
+                  if
+                    Uri.equal Semgrep_envvars.v.semgrep_url
+                      (Uri.of_string "https://semgrep.dev")
+                  then Fmt.string ppf ""
+                  else
+                    Fmt.pf ppf " at %a"
+                      Fmt.(styled `Bold string)
+                      (Uri.to_string Semgrep_envvars.v.semgrep_url)
+                in
+                Logs.app (fun m ->
+                    m "  Fetching configuration from Semgrep Cloud Platform%a"
+                      at_url_maybe ());
+                match
```

Comment:
This is quite deeply nested code. Maybe this could be split in a separate function.

Reply:
done in [66c9692](https://github.com/returntocorp/semgrep/pull/8149/commits/66c9692fe5ec4db227b46887b15e7b18f66cf8e5)